From 80b86fc349a67ab3935591fa3d209a99c1882d56 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sat, 25 May 2024 15:55:47 +0200
Subject: [PATCH] board: solidrun: am64som: mux ethernet phy reset signals
 input-only

The DP83869 phys have a chance to lock up if reset gpio is used close to
power-on and board reset.
Update the pinmux to input-only to strongly enforce that these signals
are left floating at all times.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm/dts/k3-am642-sr-som.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/dts/k3-am642-sr-som.dtsi b/arch/arm/dts/k3-am642-sr-som.dtsi
index 4f1e67ed..2af20bac 100644
--- a/arch/arm/dts/k3-am642-sr-som.dtsi
+++ b/arch/arm/dts/k3-am642-sr-som.dtsi
@@ -283,7 +283,7 @@
 	ethernet_phy0_pins_default: ethernet-phy0-pins-default {
 		pinctrl-single,pins = <
 			/* reset */
-			AM64X_IOPAD(0x0154, PIN_OUTPUT, 7)	/* Pad PRG1_PRU1_GPO19 Mux GPIO0_84 */
+			AM64X_IOPAD(0x0154, PIN_INPUT, 7)	/* Pad PRG1_PRU1_GPO19 Mux GPIO0_84 */
 			/* reference clock */
 			AM64X_IOPAD(0x0274, PIN_OUTPUT, 5)	/* Pad EXT_REFCLK1 Mux CLKOUT0 */
 		>;
@@ -292,7 +292,7 @@
 	ethernet_phy1_pins_default: ethernet-phy1-pins-default {
 		pinctrl-single,pins = <
 			/* reset */
-			AM64X_IOPAD(0x0150, PIN_OUTPUT, 7)	/* Pad PRG1_PRU1_GPO18 Mux GPIO0_20 */
+			AM64X_IOPAD(0x0150, PIN_INPUT, 7)	/* Pad PRG1_PRU1_GPO18 Mux GPIO0_20 */
 			/* led0, external pull-down on SoM */
 			AM64X_IOPAD(0x0128, PIN_INPUT, 7)	/* Pad PRG1_PRU1_GPO8 Mux GPIO0_73 */
 			/* led1/rxer */
@@ -303,7 +303,7 @@
 	ethernet_phy2_pins_default: ethernet-phy2-pins-default {
 		pinctrl-single,pins = <
 			/* reset */
-			AM64X_IOPAD(0x00d4, PIN_OUTPUT, 7)	/* Pad PRG1_PRU0_GPO7 Mux GPIO0_52 */
+			AM64X_IOPAD(0x00d4, PIN_INPUT, 7)	/* Pad PRG1_PRU0_GPO7 Mux GPIO0_52 */
 			/* led0, external pull-down on SoM */
 			AM64X_IOPAD(0x00d8, PIN_INPUT, 7)	/* Pad PRG1_PRU0_GPO8 Mux GPIO0_53 */
 			/* led1/rxer */
-- 
2.35.3

